std/experimental/allocator/building_blocks/ascending_page_allocator.d

```diff
@@ -101,13 +101,14 @@ public:
     {
         import std.algorithm.comparison : min;
 
+        immutable pagedBytes = numPages * pageSize;
         size_t goodSize = goodAllocSize(n);
-        if (goodSize > numPages * pageSize || offset - data > numPages * pageSize - goodSize)
+        if (goodSize > pagedBytes || offset - data > pagedBytes - goodSize)
```

Comment:
`if (goodSize > pagedBytes || offset + goodSize > data + pagedBytes)` would be better from a defensive viewpoint (no subtraction of unsigned values) but you decide whether it's at all sensible :)

Reply:
When I did this, I was afraid that some sort of an overflow could happen.
By doing substraction and knowing that ``offset > data`` and ``pagedBytes > goodSize`` I didnt have to do any overflow checks.